vsintegration/src/FSharp.LanguageService/ProjectSitesAndFiles.fs

```diff
@@ -231,93 +235,120 @@ type internal ProjectSitesAndFiles() =
+                        if not (isNull project) then
+                            for reference in project.ProjectReferences do
+                                let project = workspace.CurrentSolution.GetProject(reference.ProjectId)
+                                let siteProvider = provideProjectSiteProvider (workspace, project, serviceProvider, projectOptionsTable)
+                                let outputPath =
+                                    if not (String.IsNullOrWhiteSpace(project.OutputFilePath)) then
+                                        project.OutputFilePath
+                                    else
+                                        // ======================================================================
+                                        // Giant Hack here !!!!!!
+                                        // project.OutputFilePath does not have a value at this point ... 
+                                        // so calculate one by peeking at the command line options ... --out:
+                                        // ======================================================================
+                                        match projectOptionsTable with
+                                        | None -> ""
+                                        | Some projectOptions ->
+                                            let _, _, (options:string[]) = projectOptions.GetCommandLineOptionsWithProjectId(reference.ProjectId)
+                                            let outpath =
+                                                match options |> Seq.tryFind(fun opt -> opt.StartsWith("--out:") || opt.StartsWith("-o:")) with
+                                                | Some s when s.StartsWith("--out:") -> s.Substring(6)
+                                                | Some s when s.StartsWith("-o:") -> s.Substring(3)
+                                                | _ -> ""
+                                            match outpath with
+                                            | "" -> ""
+                                            | s when Path.IsPathRooted(s) -> s
+                                            | s -> Path.Combine(Path.GetDirectoryName(project.FilePath), s)
+                                // ======================================================================
+                                // Giant Hack here !!!!!!
+                                // For reasons I don't even begin to understand yet FCS needs the bin output directory
+                                // not the intermediate output directory.
+                                // so as a Giant kludge transform obj to bin
+                                // ======================================================================
+                                yield Some projectId, project.FilePath, outputPath.Replace(@"\obj\", @"\bin\"), siteProvider
```

Comment:
Yeah, this won't work in general.  Both of these paths can (and frequently are) changed by msbuild variables.  In Roslyn for non-CPS projects, we get this from the hierarchy with [this code](http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/ProjectSystem/Legacy/AbstractLegacyProject.cs,caef6e8b018c1b00).  Not sure if that would work for CPS or not. (Tag @davkean)

Reply:
I know it won't that's why the 'Hack!!!!!!' I think I am pretty sure I need the project System to have set this value.  Or I should figure out why FCS needs it ... which is probably more of a winner here.